## Multiple Linear Regression using SKLEARN

### Prdict target variable using multiple features

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [2]:
#Reading the data
cars = pd.read_csv('cars.csv')
X = cars[['Cylinders','Weight','Horsepower']]
y = cars[['MPG']]

In [3]:
from sklearn.model_selection import train_test_split

# Split the data into 80% train and 20% test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

In [4]:
from sklearn.linear_model import LinearRegression

regressor = LinearRegression()

#fitting the model
regressor.fit(X_train, y_train)

LinearRegression()

In [10]:
# get coefficients of the features and y intercept
print(regressor.coef_)
print(regressor.intercept_)

[[-0.78473746 -0.00474677 -0.03746656]]
[45.31454491]


In [5]:
# Make prediction for the test data
y_pred = regressor.predict(X_test)

In [11]:
#Show predicted data
np.round(y_pred,3)

array([[26.539],
       [12.877],
       [31.324],
       [13.779],
       [23.089],
       [14.714],
       [30.044],
       [11.388],
       [29.008],
       [ 7.105],
       [31.672],
       [24.951],
       [27.726],
       [28.614],
       [14.156],
       [21.024],
       [29.087],
       [28.963],
       [30.157],
       [28.906],
       [29.725],
       [14.424],
       [19.522],
       [19.245],
       [21.553],
       [20.578],
       [29.725],
       [15.799],
       [15.755],
       [20.275],
       [18.912],
       [24.763],
       [24.525],
       [ 9.252],
       [28.04 ],
       [11.89 ],
       [29.727],
       [29.151],
       [23.027],
       [21.281],
       [27.51 ],
       [22.88 ],
       [21.847],
       [30.172],
       [10.127],
       [15.325],
       [30.088],
       [17.241],
       [29.526],
       [27.175],
       [11.112],
       [25.872],
       [23.274],
       [31.478],
       [11.776],
       [13.305],
       [20.894],
       [29.49 ],
       [14.228

In [12]:
from sklearn import metrics
import math

mse = metrics.mean_squared_error(y_test, y_pred)
rmse = math.sqrt(mse)
print("RMSE: ", rmse)

print("Target range: ", y_test.min()," to ", y_test.max())

RMSE:  6.579444713771408
Target range:  MPG    0.0
dtype: float64  to  MPG    46.6
dtype: float64


### Root mean squared error of model is 6.57, which is good, since range is 0 to 46.6